Лаб2

T1t0-0 opened the pull request:
Не получилось сделать метод get_index_by_book_id, ошибка: "TypeError: "Book" is not subscriptable". Я думал решить это использованием метода .parse_obj в начале, но не удается импортировать pydantic, хотя исходя из данных консоли все установлено(скрины прилагаю), подскажите, что может быть не так?
![изображение](https://github.com/user-attachments/assets/2e56fa04-52d7-4340-ac21-a91017be2d0b)
![изображение](https://github.com/user-attachments/assets/0ad86670-3646-4b01-bdce-38b0ab6f1ad4)
![изображение](https://github.com/user-attachments/assets/4eacd1d3-6478-47fa-a4dc-bacd3fbcbaf0)

T1t0-0 committed:
Add files via upload

Не получилось сделать метод get_index_by_book_id, ошибка: "TypeError: "Book" is not subscriptable". Я думал решить это использованием метода .parse_obj в начале, но не удается импортировать pydantic, хотя исходя из данных консоли все установлено(скрины прилагаю), подскажите, что может быть не так?

TimurIslamgulov commented on main2.py:
```diff
@@ -0,0 +1,56 @@
+
+    def get_index_by_book_id(self, id_:int):
+        for book in self.books:
+            if book["id_"] is id_:
```
но book же это не словарь а объект, надо обращаться через точку.

T1t0-0 committed:
Update main2.py

Подскажите причину такой низкой оценки? Можете что то подсказать по вышеописанной проблеме с pydantic?
